FEC-10993 SmartSwitch Enablement

GouravSna opened the pull request.

GouravSna committed:
- SmartSwitch Config

GouravSna committed:
- SmartSwitch Configuration

GouravSna committed:
- Travis and version file

GouravSna committed:
- Travis build

GouravSna committed:
- Travis build

GouravSna committed:
- Rename package

GouravSna committed:
- Package name change

GouravSna committed:
- Package name change

GouravSna committed:
- Added gradle build files

GouravSna committed:
- Added missing files

GouravSna committed:
- Added gitignore

GouravSna committed:
- Code Refactor

GouravSna committed:
- Converted Java class to Kotlin class
- Code refactor

giladna committed:
move to maven first repo uploaded

giladna committed:
maven stuff

giladna committed:
gradle stuff

giladna committed:
SMARTSITCH_ERROR

giladna committed:
SMARTSWITCH_ERROR

GouravSna committed:
- Code refactor

GouravSna committed:
- Added .kt exclusion in mvn script

GouravSna committed:
- Gradle update

GouravSna committed:
- Replaced with Snapshot

GouravSna committed:
- Null checks
- Terminate service

GouravSna committed:
- Added doc

GouravSna committed:
- Code Refactor

GouravSna committed:
- Empty Line

GouravSna committed:
- Code refactor

GouravSna committed:
- Code refactor
- Made SmartSwitch Server url as optional in Config

giladna committed:
update link

https://github.com/kaltura/playkit-android-smartSwitch/blob/develop/LICENSE

giladna committed:
Update SmartSwitchExecutor.kt

giladna commented on smartswitchplugin/src/main/java/com/kaltura/playkit/plugins/smartswitch/SmartSwitchExecutor.kt:
```diff
@@ -0,0 +1,180 @@
+            val inputStream: InputStream?
+            var smartSwitchUri: Uri
+            try {
+                smartSwitchUri = Uri.parse(smartSwitchServerUrl)
```
@GouravSna 

maybe call this  in 1 line?

smartSwitchUri = appendQueryParams(Uri.parse(smartSwitchServerUrl))

GouravSna replied:
Done

giladna commented on smartswitchplugin/src/main/java/com/kaltura/playkit/plugins/smartswitch/SmartSwitchExecutor.kt:
```diff
@@ -0,0 +1,180 @@
+                }
+                return error(resourceUrl, errorMessage)
+            } finally {
+                connection?.disconnect()
```
@GouravSna 

please call  inputStream.close()  in finally 
I think it is also required

GouravSna replied:
Done

GouravSna committed:
- Code refactor
- Closing the connection resources

GouravSna committed:
- Code refactor
- Returning null for error response in case we got the response from SmartSwitch server

GouravSna committed:
- Empty line

GouravSna committed:
- Removed Empty line
